src/client/components/responsive-video/responsive-video.vue

```diff
@@ -1,17 +1,113 @@
 <template>
-  <figure>
-    <a :href="video.url" target="_blank" rel="noopener" @click="play()">
-      <img alt="" :src="video.thumbnailUrl" :width="video.width" :height="video.height" />
-    </a>
-    <figcaption v-if="video.title">{{ video.title }}</figcaption>
-  </figure>
+  <div class="responsive-video">
+    <figure>
+      <fixed-ratio class="responsive-video__canvas" :width="video.width" :height="video.height">
+        <lazy-load>
+          <img
+            class="responsive-video__image"
+            v-if="!isPlaying"
+            alt=""
+            :src="video.thumbnailUrl"
+          />
+        </lazy-load>
+        <iframe
+          v-if="isPlaying"
+          :src="videoSource"
+          style="width:100%; height:100%;"
+          frameborder="0"
+          webkitallowfullscreen
+          mozallowfullscreen
+          allowfullscreen
+          allow="autoplay">
+        </iframe>
+        <a :href="video.url" v-if="!isPlaying" class="responsive-video__button" @click="play()">
```

Comment:
Put `img.responsive-video__image` inside this element? To me it's one big clickable item which disappears entirely the moment the video `isPlaying`.